webrtc.html

```diff
@@ -2582,6 +2636,116 @@
       </dl>
     </section>
 
+    <section>
+      <h2 id="sec.cert-mgmt">Certificate Management</h2>
+      <p>
+        The certificates that <code>RTCPeerConnection</code> instances use to
+        authenticate with peers use the <a><code>RTCCertificate</code></a>
+        interface.  These objects can be explicitly generated by applications
+        using <code>RTCPeerConnection.generateCertificate</code> and provided in
+        the <a><code>RTCConfiguration</code></a> when constructing a
+        new <code>RTCPeerConnection</code> instance.
+      </p>
+      <p>
+        The explicit certificate management functions provided here are
+        optional.  If an application does not provide
+        the <code>certificates</code> configuration option when constructing
+        an <code>RTCPeerConnection</code> a new set of certificates MUST be
+        generated by the <a>user agent</a>.  That set MUST include an ECDSA
+        certificate with a private key on the P-256 curve and a signature with a
+        SHA-256 hash.
+      </p>
+
+      <dl class="idl" title="partial interface RTCPeerConnection">
+        <dt>static Promise&lt;RTCCertificate&gt; generateCertificate
+          (AlgorithmIdentifier keygenAlgorithm)</dt>
+        <dd>
+          <p>This function takes an argument that describes how the private key associated with the certificate is generated.  The <var>keygenAlgorithm</var> argument uses the WebCrypto
+          [[!WebCryptoAPI]] <a href="https://dvcs.w3.org/hg/webcrypto-api/raw-file/tip/spec/Overview.html#dfn-AlgorithmIdentifier">AlgorithmIdentifier</a>
+          type.  The <var>keygenAlgorithm</var> value MUST be a valid argument
+            to <a href="https://dvcs.w3.org/hg/webcrypto-api/raw-file/tip/spec/Overview.html#SubtleCrypto-method-generateKey"><code>Crypto.subtle.generateKey</code></a>.</p>
+
+          <p>The algorithm name used in the <code>AlgorithmIdentifier</code>
+          MUST be valid for both key generation and signing.
+          The <var>keygenAlgorithm</var> value MUST produce a non-error result
+          when normalized according to the
+          WebCrypto <a href="https://dvcs.w3.org/hg/webcrypto-api/raw-file/tip/spec/Overview.html#algorithm-normalization">algorithm
+          normalization process</a> [[!WebCryptoAPI]].  The following algorithm
```

Comment:
As noted in the link you provided, and what I specified earlier, algorithm normalization requires both an input and a desired operation.

Line 2669 says "MUST be valid for both key generation and signing".

Possible interpretations:
1) Normalize for generateKey
2) Normalize for sign
3) Normalize for generateKey and sign
4) Normalize for some other operation
5) Just convert DOMString inputs to { name: "foo" } form

Only due to following this PR, not due to the proposed language itself, I suspect you want 1.
However, from following this PR, it's unclear to me whether or not it is *possible* to request a 512-bit key. Are you just using the names, or are you also using the generateKey parameters in full?

If using the generateKey parameters, then it's unclear if it's valid to request a 512-bit RSA key. Or an even exponent. Or a non-prime exponent. or e=3 when the browser really, really, really hates non-F4.